### DISCUSSION TITLE
Bot keyboard keeps dissapearing

### DISCUSSION THREAD

**Discussion by simathedev** [telegraf/telegraf#1795] [Q&A]
Hey, Could please help me I'm encountering problems where my keyboard keeps dissapearing and I'm required to keep pressing a square icon button to display it again.

I've tried changing the one_time_keyboard to "false" but I still get the same result
```
 const requestRestartMenuKeyboard={
    "reply_markup": {
        "one_time_keyboard": false,
        "resize_keyboard":true,
          "keyboard":[
            [{
                text:"Login",
                one_time_keyboard:false,
                resize_keyboard:true

            }],
        ]
    }
  };
```
My code does request the keyboards:

`ctx.replyWithHTML("Operation cancelled.\n\nAmount entered is not a number",requestRestartMenuKeyboard);`

I really don't know what could be causing this and it's a recent thing.

**Comment by MKRhere** [comment]
This looks fine; one_time_keyboard and resize_keyboard can only be specified at the top level and not for each button. Instead of writing the keyboard by hand, you may want to use the Markup helper:

```TS
const requestRestartMenu = Markup.keyboard([Markup.button.text("Login")])
		.resize(true)
		.oneTime(false); // this is not really necessary, the default is false

ctx.reply("Operation cancelled.\n\nAmount entered is not a number", requestRestartMenu);
```

It's less error-prone this way. The client may also choose to collapse the keyboard for any other reason. This is the most control we have.

**Reply by simathedev** [reply]
Thank you for the reply @MKRhere . I'm still experiencing the same problem.

I've noticed that whenever a user enters input and sends it to the bot, the I-cursor/I-beam remains inside the telegram message input field. This causes the bot keyboard not to automatically "popup" as it should.

Is there any way I can remove the focus on the input field, to remove the I-cursor so the keyboard can appear?

**Reply by MKRhere** [reply]
Nope, we have no control over this. Sorry!
